libvast/src/transform_steps/project.cpp

```diff
@@ -65,7 +65,6 @@ project_step::add(type layout, std::shared_ptr<arrow::RecordBatch> batch) {
       transformed_.clear();
       return layout_result.error();
     }
-    transformed_.emplace_back(layout, std::move(batch));
```

Comment:
This needs a changelog entry. I suggest the following:

> The `project` transform step now correctly removes data where no configured columns match instead of leaving the data unchanged.